File: java/p_0002.java
/*
 * Euler problem 2
 * By considering the terms in the Fibonacci sequence whose values do not
 * exceed four million, find the sum of the even-valued terms.
 *
 */

public class p_0002 {
  private static long sum_even_fibonacci(long limit) {
    long total = 0L;
    long first = 1L;
    long second = 2L;

    while (second < limit) {
      if (second % 2 == 0) {
        total += second;
      }

      long next = first + second;
      first = second;
      second = next;
    }

    return total;
  }

  public static void main(String [] args) {
    System.out.printf("The sum is %d\n", sum_even_fibonacci(4000000L));
  }
}

File: cpp/src/p_0002.cc
/*
 * By considering the terms in the Fibonacci sequence whose values do not
 * exceed four million, find the sum of the even-valued terms.
 */

#include "euler.h"

int euler_0002(int limit) {
    // Return the sum of the even-valued terms.
    int total{};
    int first{1};
    int second{2};
    while (second < limit) {
        if (second % 2 == 0) total += second;
        int temp{second};
        second += first;
        first = temp;
    }
    return total;
}
